### ISSUE TITLE
[Enhancement] Flutter Sound on Flutter Web

### ISSUE THREAD

**Issue by Larpoux** [dooboolab/flutter_sound#468]
We want Flutter Sound running on Flutter Web.

I am currently working on this new feature.
Maybe in 6.2 or 6.3.  Perhaps 🙄 ...

**Comment by ffeldhaus** [comment]
Implementing this for web should be relatively straight forward using `dart:html` and [`getUserMedia`](https://api.dart.dev/stable/2.10.2/dart-html/Navigator/getUserMedia.html) together with [`AudioElement`](https://api.dart.ranyunlong.com/dart-html/AudioElement-class.html).

I just opened an [issue for the Dart documentation that the example for `getUserMedia` should be improved](https://github.com/dart-lang/site-www/issues/2701). The issue was regarding the VideoElement, but also for the AudioElement, the stream should be directly passed to `srcObject` instead of using `src`. This is a dart example to record audio to stream and immediately play it back via AudioElement:

```
window.navigator.getUserMedia(audio: true).then((stream) {
  var audio = new AudioElement()
    ..autoplay = true
    ..srcObject = stream;
  document.body.append(audio);
```

**Comment by Larpoux** [comment]
Thank you for this info, @ffeldhaus .
This is exactly the kind of valuable info which arrives at **the good time**. 👍 
I have already created a flutter_sound_interface in V6.1 so that it will be easier to support flutter web.
But I have not started to work on Flutter Sound for Flutter Web, because I wanted to re-structure flutter sound before, so that the code will be isolated from flutter. It will be V6.2.

Flutter Sound on Flutter Web will be V6.3

**Comment by ffeldhaus** [comment]
The code from the [Fluter VideoPlayer could probably be a good reference as it is already available for web](https://github.com/flutter/plugins/blob/master/packages/video_player/video_player_web/lib/video_player_web.dart).

**Comment by ffeldhaus** [comment]
After trying to implement something working all evening, I'm struggling most with recording. It seems using [`MediaRecorder`](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) is a good idea. It ensures that data is cached and delivered via an event. In the end I was struggling to get the data into a dart stream.

Here is some input:
```dart
MediaRecorder mediaRecorder;
# add stream from audio device to MediaRecorder, ideally the input device source can be selected
window.navigator.getUserMedia(audio: true).then((stream) {
      mediaRecorder = new MediaRecorder(stream);    
    });
# start recording and deliver the data every 10ms
mediaRecorder.start(10);
# register event listener to collect data
mediaRecorder.addEventListener('dataavailable', (Event event) {
        print("datavailable ${event.runtimeType}");
        final Blob blob = JsObject.fromBrowserObject(event)['data'];
        print("blob size: ${blob.size}");
    });
# stop recording
mediaRecorder.stop();
```

**Comment by Larpoux** [comment]
Thank you @ffeldhaus .
Your input is really very useful for me. 👍 
Your investigations will spare me much time.
(of course if you have some pull request, which can be incorporated inside Flutter Sound do not hesitate.).

I re-open [#297](https://github.com/dooboolab/flutter_sound/issues/297) because @hyochan had some ideas for Flutter Sound on Web. Maybe you can have a look to this very old post.

**Comment by Larpoux** [comment]
Will be next version (6.4)

**Comment by Larpoux** [comment]
Feature implemented in V6.4

**Comment by CrimsonPiik** [comment]
> After trying to implement something working all evening, I'm struggling most with recording. It seems using [`MediaRecorder`](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) is a good idea. It ensures that data is cached and delivered via an event. In the end I was struggling to get the data into a dart stream.
> 
> Here is some input:
> 
> ```dart
> MediaRecorder mediaRecorder;
> # add stream from audio device to MediaRecorder, ideally the input device source can be selected
> window.navigator.getUserMedia(audio: true).then((stream) {
>       mediaRecorder = new MediaRecorder(stream);    
>     });
> # start recording and deliver the data every 10ms
> mediaRecorder.start(10);
> # register event listener to collect data
> mediaRecorder.addEventListener('dataavailable', (Event event) {
>         print("datavailable ${event.runtimeType}");
>         final Blob blob = JsObject.fromBrowserObject(event)['data'];
>         print("blob size: ${blob.size}");
>     });
> # stop recording
> mediaRecorder.stop();
> ```

// MY EDITS FOR ( IPHONE DEVICES ) AFTER DEPLOYMENT instead of asking for microphone permissions
      var constraints = {
        'audio': true,
      };
      html.window.navigator.mediaDevices!
          .getUserMedia(constraints)
          .then((stream) {.....}
